File: Easy Challenges/Challenge #0001 [Easy]/solutions/solution.rb
if File.exist?('userinfo.txt')
  name, age, username = File.readlines('userinfo.txt').map(&:chomp)
  puts "Your name is #{name}, you are #{age} years old, and your username is #{username}"
else
  print 'Your name: '
  name = gets
  print 'Your age: '
  age = gets
  print 'Your username: '
  username = gets
  File.open('userinfo.txt', 'w') do |f|
    puts "Your name is #{name}, you are #{age} years old, and your username is #{username}"
    f.puts [name, age, username]
  end
end

File: Easy Challenges/Challenge #0001 [Easy]/solutions/solution.py
def data_parser(data):
	print 'Your name is {0}, you are {1} years old, and your username is {2}.'.format(data[0],data[1],data[2])
	if 'y' in data[3]:
		f = open('data_logger.txt','w')
		f.write('Your name is {0}, you are {1} years old, and your username is {2}.'.format(data[0],data[1],data[2]))

def data_retriever():
	save_prompt = input('Save this data to disk, or no?\n>')
	name_input = input('What\'s your name?\n>')
	age_input = input('What\'s your age?\n>')
	reddit_username_input = input('What\'s your reddit username?\n>')
	return name_input,age_input,reddit_username_input,save_prompt

data_parser(data_retriever())
